python/lsst/analysis/drp/histPlot.py

```diff
@@ -274,94 +236,95 @@ def histPlot(self, catPlot, plotInfo, sumStats):
+
+        # panel plotting
+        for i, (panel, col_start, col_stop, row_start, row_stop) in enumerate(zip(panels,
+                                                                                  col_starts, col_stops,
+                                                                                  row_starts, row_stops)):
+            hists = [x for x in panels[panel].toDict()['actions'].keys()]
+            columns = []
+            for hist in hists:
+                columns.append(f"p{i}_{hist}")
+
+            # panel plotting limits
+            vLowers, vUppers = [], []
+            for column in columns:
+                pvals = np.nanpercentile(datPlot[column].values,
+                                         [self.config.panels[panel].pLower, self.config.panels[panel].pUpper])
+                vLowers.append(pvals[0])
+                vUppers.append(pvals[1])
+            vLower = np.min(vLowers)
+            vUpper = np.max(vUppers)
+
+            # generate plot
+            ax = fig.add_subplot(gs[row_start:row_stop, col_start:col_stop])
+            for count, (hist, column) in enumerate(zip(hists, columns)):
+                isfinite = np.isfinite(datPlot[column])
+                datMed = np.nanmedian(datPlot[column][isfinite])
+                datMad = sigmaMad(datPlot[column][isfinite])
+                col = plt.cm.tab10(count)
+                histLabel = f"{hist}:::{datMed:0.1f}, {datMad:0.1f}, {np.sum(isfinite)}"
+                ax.hist(datPlot[column][isfinite], bins=self.config.panels[panel].nBins, alpha=0.7, color=col,
+                        range=(vLower, vUpper), histtype="bar", lw=2, label=histLabel)
+                ax.axvline(datMed, ls=':', lw=2, c=col)
+            ax.set_yscale(self.config.panels[panel].yscale)
+            ax.set_xlim(vLower, vUpper)
+            ylims = ax.get_ylim()
+            ax.set_ylim(ylims[0], ylims[1]*1.1)
+            ax.set_xlabel(self.config.panels[panel].label, labelpad=1)
+            ax.tick_params(labelsize=7)
+            # add legends
+            leg1 = ax.legend(loc="upper left", fontsize=7, frameon=False, handlelength=0, borderaxespad=0,
+                             handletextpad=0, handleheight=0.7, framealpha=0.4)
+            _ = plt.gca().add_artist(leg1)
+            for count, text in enumerate(leg1.get_texts()):
+                text.set_text(text.get_text().split(":::")[0])
+                text.set_color(plt.cm.tab10(count))
+            leg2 = ax.legend(loc="upper right", fontsize=7, frameon=False, handlelength=0, borderaxespad=0,
+                             handletextpad=0, handleheight=0.7, framealpha=0.4)
+            for count, text in enumerate(leg2.get_texts()):
+                text.set_text(text.get_text().split(":::")[1])
+                text.set_color(plt.cm.tab10(count))
+            renderer = fig.canvas.get_renderer()
+            tlens = [t.get_window_extent(renderer).width for t in leg2.get_texts()]
```

Comment:
Some comments here, and in other places, would be useful to know what is going on.

Reply:
This section has been simplified. Additionally, a number of in-line comments have been added explaining the various steps of data processing.